goscraper.go

```diff
@@ -66,6 +128,23 @@ func (scraper *Scraper) Scrape() (*Document, error) {
 	return doc, nil
 }
 
+func (scraper *Scraper) GetDocument() (*Document, error) {
+	doc, err := scraper.getDocument()
+	if err != nil {
+		return nil, err
+	}
+
+	return doc, nil
+}
+
+func (scraper *Scraper) ParseDocument(doc *Document) (*Document, error) {
+	err := scraper.parseDocument(doc)
```

Comment:
same here, maybe `return scraper.parseDocument` will be an easier way to handle it?